albumentations/augmentations/transforms.py

```diff
@@ -792,8 +795,35 @@ def get_params_dependent_on_targets(self, params):
 
         holes = []
         for _n in range(random.randint(self.min_holes, self.max_holes)):
-            hole_height = random.randint(self.min_height, self.max_height)
-            hole_width = random.randint(self.min_width, self.max_width)
+            if all(
+                [
+                    isinstance(self.min_height, int),
+                    isinstance(self.min_width, int),
+                    isinstance(self.max_height, int),
+                    isinstance(self.max_width, int),
+                ]
+            ):
+                hole_height = random.randint(self.min_height, self.max_height)
+                hole_width = random.randint(self.min_width, self.max_width)
+            elif all(
+                [
+                    isinstance(self.min_height, float),
+                    isinstance(self.min_width, float),
+                    isinstance(self.max_height, float),
+                    isinstance(self.max_width, float),
+                ]
+            ):
+                hole_height = random.randint(round(self.min_height * height), round(self.max_height * height))
+                hole_width = random.randint(round(self.min_width * width), round(self.max_width * width))
```

Comment:
I think is better to change to these lines:
```python
hole_height = height * random.uniform(self.min_height, self.max_height)
hole_width = width * random.uniform(self.min_width, self.max_width)
```

Reply:
Made the changes